docs/sequence/PISPTransactionApi.puml

```diff
@@ -27,40 +38,34 @@ rnote right of PISP #Light
   }
 }
 end note
-PISP -> OBS: ""POST /thirdpartyTransactions""
+PISP -> 3p_PISP_OBS: POST /thirdpartyTransactions
 activate PISP
-activate OBS
+activate 3p_PISP_OBS
 ' TODO: line up with state-models.md doc
-OBS -> ST8: ""partyLookup()""
-rnote right of ST8: ""state: requestPartyLookup""
+3p_PISP_OBS -> ST8: partyLookup()
+rnote right of ST8: state: requestPartyLookup
 activate ST8
 
-ST8 -> Switch: ""GET /parties/{type}/{id}""
-Switch --> ST8: ""202 Accepted""
-rnote right of ST8: ""state: pendingPartyLookup""
-
-note over Switch
-  Switch performs party lookup... just like normal
-end note
-
-Switch -> IBS: ""PUT /parties/{type}/{id}""
-IBS -> ST8: ""onPartyLookup()""
-IBS --> Switch: ""200 OK""
-rnote right of ST8: ""state: payeeResolved""
+ST8 -> SDK_PISP_OBS: SYNC GET /parties/{type}/{id}
+activate SDK_PISP_OBS
+SDK_PISP_OBS -> SDK_PISP_OBS: retrieve PartiesInformation from Switch
+SDK_PISP_OBS -> ST8: 200 PartyInformation
+deactivate SDK_PISP_OBS
+rnote right of ST8: state: payeeResolved
 
-ST8 -> OBS: ""something...""
+ST8 -> 3p_PISP_OBS: partyInformation
 deactivate ST8
 
 ' TODO: party lookup details
-OBS --> PISP: ""200 OK""
-deactivate OBS
+3p_PISP_OBS --> PISP: 200 OK PartyInformation
+deactivate 3p_PISP_OBS
 deactivate PISP
 
-...User checks the party name, and confirms...
+...User checks the party name, and confirms transaction to continue...
 
 == Initate Transaction ==
 rnote right of PISP #Light
-""POST /thirdpartyTransactions/{id}/initiate""
+POST /thirdpartyTransactions/{id}/initiate
 {
   sourceAccountId: "dfspa.alice.1234",
   consentId: "555",
```

Comment:
We no longer need to provide the `consentId` here - that information is contained by the `sourceAccountId`

(see: https://raw.githubusercontent.com/mojaloop/pisp/master/docs/out/transfer/api_calls_detailed_2.svg for more information)

Reply:
@lewisdaly if we drop `consentId` from this payload (and from `POST /thirdpartyRequests/transactions` send to DFSP), than DFSP needs to resolve `sourceAccountId (accountId)` to get `consentId`. DFSP needs to use a `Map[accountId] -> consentId` for this. 

I've checked our Auth-Service and I see that for given `consentId` we are storing `sourceAccountId` on the `scopes` list - so DFSP has all data needed for this mapping. But there is no interface exposed to make such a query on Auth-Service... 

I find out that `Map[accountId] -> consentId` mapping and quering isn't trival. We can have multiple `consentId` registered for the same `accountId` 

Because from the specification of `/accounts/{UserId}` I see that we are having a list of available accounts for the user for the given DFSP. This list is used during linking a device with an account and represented in the system by Consent and its id. And one of these accounts will be used as `sourceAccountId` in our payload.

In the case when the user has multiple devices (old & new phone for example) we would have registered more than one Consent for the same AccountId. 

So the map is now `Map[accountId] -> [consentIdOldPhone, consentIdNewPhone]` 

It brings us the problem - How DFSP will recognize which `consentId` from the mapping should it use to validate POST /thirdpartyRequests/transactions ?

Reply:
Thanks for this careful exposition, @eoln.

>than DFSP needs to resolve byself sourceAccountId (accountId) to get consentId
We are also dropping the `sourceAccountId`. In the latest API Draft, Michael removed `sourceAccountId` and `consentId`, in favour of that information being contained in the payer `PartyIdInfo`. When a link is created, for each account that is linked, a `THIRD_PARTY_LINK` identifier type will be registered at the ALS, and the DFSP will be able to infer (1) the source of funds account and (2) the `consentId` based on the `payer`field alone:

<img width="482" alt="Screen Shot 2021-03-08 at 8 35 34 pm" src="https://user-images.githubusercontent.com/1683980/110306570-e5bae500-804d-11eb-932c-af317604ada0.png">

ref: https://raw.githubusercontent.com/mojaloop/pisp/master/docs/out/transfer/api_calls_detailed_2.svg

An `accountId` should be unique for a User, Device, PISP and DFSP, so it shouldn't be reused by a DFSP for a subsequent linking request with a different device.

As a part of the linking process, we expect the DFSP to keep a record of the `Map[accountId] -> consentId` mapping... perhaps for your implemention now this will live in the `mojaloop-simulator`? Or could we reasonably expect the `thirdparty-scheme-adapter` to remember this type of information?

Reply:
I've delegated the task to find out the Consent and verify it to DFSP backend.verifyConsent(pispId, payer)